<launch>

  <!-- GPS conversion : Lat/Lon to UTM X/Y -->
  <!-- from sensor_msgs/NavSatFix to gps_common/GPSFix (old GPS msg) -->
  <node name="navsatfix_to_gpsfix" pkg="gps_conversions" type="navsatfix_to_gpsfix" output="screen">
    <param name="magnetic_dip" value="0."/>
    <remap from="nav_sat_fix" to="fix"/>
  </node>
  <!-- from novariant_capi/PositionGps to gps_common/GPSFix (new GPS msg) -->
  <node name="gps_fix_conversion" pkg="novariant_capi" type="gps_fix_conversion" output="screen">
    <param name="err_position_horz" value="0.1"/>
    <param name="err_position_vert" value="0.2"/>
    <param name="err_orientation_horz" value="3.14159"/>
    <param name="err_orientation_vert" value="0.08"/>
    <param name="magnetic_dip" value="0."/>
  </node>
  <!-- from gps_common/GPSFix to nav_msgs/Odometry -->
  <node name="gps_to_utm" pkg="gps_conversions" type="gps_to_utm" output="screen">
    <param name="frame_id" value="utm"/>
    <param name="child_frame_id" value="gps"/>
    <param name="map_frame_id" value="map"/>
    <param name="tf_offset_to_map" value="true"/>
    <param name="origin_on_first_pose" value="true"/>
  </node>

  <!-- SLAM to WGS84 conversion : UTM X/Y to GPS Lat/Lon -->
  <node name="slam_to_gps" pkg="gps_conversions" type="odom_to_gps" output="screen">
    <param name="utm_frame_id" value="utm"/>
    <remap from="odom" to="slam_odom"/>
    <remap from="fix" to="slam_fix"/>
  </node>

</launch>